platform/features/layout/src/LayoutController.js

```diff
@@ -293,9 +344,102 @@ define(
          * view configuration.
          */
         LayoutController.prototype.endDrag = function () {
+            this.frameMoved = true;
+
+            setTimeout(function () {
+                this.frameMoved = false;
+            }.bind(this), 0);
+
             this.endDragInScope();
         };
 
+        /**
+         * Check if the object is currently selected.
+         *
+         * @param {string} obj the object to check for selection
+         * @returns {boolean} true if selected, otherwise false
+         */
+        LayoutController.prototype.selected = function (obj) {
+            return !!this.selectedId && this.selectedId === obj.getId();
+        };
+
+        /**
+         * Set the active user selection in this view.
+         *
+         * @param event the mouse event
+         * @param {string} id the object id
+         */
+        LayoutController.prototype.select = function (event, id) {
+            if (event) {
+                event.stopPropagation();
+                if (this.selection) {
+                    event.preventDefault();
+                }
+            }
+
+            var self = this;
+            var selectedObj = {};
+            var configuration = this.$scope.configuration;
+
+            this.selectedId = id;
+
+            // Toggle the visibility of the object frame
+            function toggle() {
```

Comment:
@larkin I can rename the function to be more descriptive. I'll remove deselect before selecting as it's done in selection.select(). Moving toggle to the prototype and calling select() at the end of the method doesn't seem to work, it will get into some recursive call that never ends.